src/mir/node.rs

```diff
@@ -772,11 +775,13 @@ impl MirNodeType {
             MirNodeType::Project {
                 emit: ref our_emit,
                 literals: ref our_literals,
+                arithmetic: ref our_arithmetic,
             } => match *other {
                 MirNodeType::Project {
                     ref emit,
                     ref literals,
-                } => our_emit == emit && our_literals == literals,
+                    ref arithmetic,
+                } => our_emit == emit && our_literals == literals && our_arithmetic == arithmetic,
```

Comment:
This is quite conservative (though probably fine for v0 of arithmetic expression support): it will only allow reuse of projections that have *exactly identical* arithmetic expressions. For example, with multiple elements in the `arithmetic` vector, it would only reuse the projection if the vectors have the same elements and they're all identical. However, it is also acceptable to reuse the projection if the new `our_arithmetic` is a strict subset of `arithmetic`.

Now, you might observe that we are similarly unnecessarily strict about `emit`, and you'd be right ;-)